service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/NotificationHandlerTest.java

```diff
@@ -67,9 +75,63 @@
     @Mock
     private GatekeeperEmailContentProvider gatekeeperEmailContentProvider;
 
+    @MockBean
+    private UserDetailsService userDetailsService;
+
     @InjectMocks
     private NotificationHandler notificationHandler;
 
+    @Test
+    void shouldNotNotifyHmctsAdminOnC2Upload() throws IOException, NotificationClientException {
+        final String SUBJ_LINE = "Lastname, SACCCCCCCC5676576567";
+        final Map<String, Object> parameters = ImmutableMap.<String, Object>builder()
+            .put("subjectLine", SUBJ_LINE)
+            .put("hearingDetailsCallout", SUBJ_LINE)
+            .build();
+
+        given(userDetailsService.getUserDetails(AUTH_TOKEN))
+            .willReturn(new UserDetails("1", "hmcts-admin@test.com",
+                "Hmcts", "Test", UserRole.HMCTS_ADMIN.getRoles()));
+
+        given(hmctsEmailContentProvider.buildC2UploadNotification(callbackRequest().getCaseDetails()))
+            .willReturn(parameters);
+
+        notificationHandler.sendNotificationForC2Upload(new C2UploadNotifyEvent(callbackRequest(), AUTH_TOKEN, USER_ID,
+            userDetailsService.getUserDetails(AUTH_TOKEN)));
+
+        verify(notificationClient, times(0)).sendEmail(
+            isNull(), isNull(), isNull(), isNull());
+
+    }
+
+    @Test
+    void shouldNotifyNonHmctsAdminOnC2Upload() throws IOException, NotificationClientException {
+        final String SUBJ_LINE = "Lastname, SACCCCCCCC5676576567";
+        final Map<String, Object> parameters = ImmutableMap.<String, Object>builder()
+            .put("subjectLine", SUBJ_LINE)
+            .put("hearingDetailsCallout", SUBJ_LINE)
+            .build();
+
+        given(userDetailsService.getUserDetails(AUTH_TOKEN))
+            .willReturn(new UserDetails("1", "hmcts-non-admin@test.com",
+                "Hmcts", "Test", UserRole.LOCAL_AUTHORITY.getRoles()));
+
+        given(hmctsCourtLookupConfiguration.getCourt(LOCAL_AUTHORITY_CODE))
+            .willReturn(new Court(COURT_NAME, COURT_EMAIL_ADDRESS));
+
+        given(localAuthorityNameLookupConfiguration.getLocalAuthorityName(LOCAL_AUTHORITY_CODE))
+            .willReturn("Example Local Authority");
+
+        given(hmctsEmailContentProvider.buildC2UploadNotification(callbackRequest().getCaseDetails()))
+            .willReturn(parameters);
+
+        notificationHandler.sendNotificationForC2Upload(new C2UploadNotifyEvent(callbackRequest(), AUTH_TOKEN, USER_ID,
+            userDetailsService.getUserDetails(AUTH_TOKEN)));
+
+        verify(notificationClient, times(1)).sendEmail(
+            eq(C2_UPLOAD_SUBMISSION_TEMPLATE), notNull(), eq(parameters), eq("12345"));
```

Comment:
Couldn't we check what's the email in the second parameter? You mocked it couple lines above.

Reply:
good spot

Reply:
updated